### ISSUE TITLE
Add parental controls 

### ISSUE THREAD

**Issue by JohannHilbold** [amosyuen/ha-tplink-deco#189]
Hey guys, 
I'm wondering if it's possible to add parental control switches. In the Deco app (on Android at least) you can display information and turn on/off internet for some "device groups". Although the information is not very accurate, the toggle internet feature works pretty well. 

![image](https://github.com/amosyuen/ha-tplink-deco/assets/36189403/90e463e2-196a-4b02-9bc4-c4d4fd77ebf0)

**Describe the solution you'd like**
Ideally, I would like to expose some switches on HA to toggle internet for this or that group. 

**Describe alternatives you've considered**
Well I can use the Deco app to turn it off/on, but the idea was to automate it in HA. The HA companion app exposes an "interactive mode" boolean which says when the screen of a mobile/tablet is turned on or off. So when "kid A" has had "interactive mode" for 2 hours that day, it's time to take a break from the internet! 

**Additional context**
I'm a developper myself and although I don't know python, the few lines I've read on this repo here don't scare me that much (yet). With a bit of help and some pointers on how this addon works, I wouldnt mind trying to do it myself. 

Cheers

**Comment by amosyuen** [comment]
So this integration connects with the router web admin UI not the mobile app. I don't think I saw parental controls in the router web admin UI. If the router doesn't support it, then you would need to figure out how to reverse engineer the mobile app API and auth.

**Comment by JohannHilbold** [comment]
This is correct. The web admin UI does not display parental controls. But at the end of the day, when a parent turns off internet for their kid, it must be done at the router level. 
So all there is no official API for Deco? All this integration does is mimic the admin web UI ? it's a nice feat, but that does seem to limit the possibilities for me

**Comment by amosyuen** [comment]
Yep, pretty much can only try to reverse engineer their API which is quite time consuming. Usually much harder with mobile.

**Comment by scottmatthewandrews** [comment]
Reviving an old thread.
Short of getting rid of my M5's and going to a Unquity system, I'm motivated to at least take a look.

Even my old TP-Link ADSL modem did a better job at URL filtering and multiple time of day use!

Any thoughts or suggestions in capturing the app traffic would be helpful - but I will search around.  I'm on iOS but could install the Android version in an emulator.

Have also opened https://github.com/AlexandrErohin/home-assistant-tplink-router/issues/88
